egs/swbd/s5c/local/nnet3/prepare_multistyle_data.sh

```diff
@@ -0,0 +1,242 @@
+#!/bin/bash
+
+. ./cmd.sh
+
+set -e
+stage=1
+train_stage=-10
+generate_alignments=false
+multi_style=false
+noise_list="reverb:music:noise:babble:clean"
+augment_test_set=false
+suffix=""
+clean_ali=tri4_ali_nodup
+train_set=train_nodup
+
+. ./path.sh
+. ./utils/parse_options.sh
+
+mkdir -p exp/nnet3
+
+if [ -e data/rt03 ]; then maybe_rt03=rt03; else maybe_rt03= ; fi
+
+if [ "$multi_style" == "true" ]; then
+  suffix=_ms
+fi
+
+if [ "$multi_style" == "true" ]; then
+  if [ $stage -le 1 ]; then
```

Comment:
I think you can break this up into more stages, in case it dies somewhere in the middle.  And please have more of the stages check whether their work has already been done and die with a suitable message if so.